shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.h

```diff
@@ -10,4 +10,19 @@
 
 - (uint32_t)glFrameBufferId;
 
+- (void)setLayerContent;
+- (void)setLayerContentWithIOSurface:(IOSurfaceRef*)ioSurface;
+
+- (void)BackTextureWithIOSurface:(CGSize)size
+              withBackingTexture:(GLuint)texture
+                         withFbo:(GLuint)fbo
+                   withIOSurface:(IOSurfaceRef*)ioSurface;
+
+// Methods used by FlutterMacOSCompositor to render Flutter content
+// using a single Framebuffer/Texture/IOSurface.
+- (uint32_t)getFramebuffer;
```

Comment:
For all the following methods answer these questions in the comments.

1. What is returned given the surface manager can maintain multiple of these.
2. When are each of these called?
3. What are the lifecycles for each of these returned objects?

Reply:
Done